src/ucc/api/ucc.h

```diff
@@ -1574,5 +1588,226 @@ static inline ucc_status_t ucc_collective_test(ucc_coll_req_h request)
+ * given ee, the @ref ucc_ee_create and @ref ucc_ee_destroy must be invoked from
+ * the same thread.
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ ucc_status_t ucc_ee_destroy(ucc_ee_h ee);
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine gets the event from the event queue.
+ *
+ * @param [in]  ee        execution engine handle
+ * @param [out] ev        Event structure fetched from the event queue
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * @ref ucc_ee_get_event fetches the events from the execution engine. If there
+ * are no events posted on the ee, it returns immediately without waiting for
+ * events. All events must be acknowledged using the @ref ucc_ee_ack_event
+ * interface. The event acknowledged is destroyed by the library. An event
+ * fetched with @ref ucc_ee_get_event but not acknowledged might consume
+ * resources in the library.
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_get_event(ucc_ee_h ee, ucc_ev_t *ev);
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine acks the events from the event queue.
+ *
+ * @param [in]  ee      execution engine handle
+ * @param [in]  ev      Event to be acked
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * An event acknowledged by the user using @ref ucc_ee_ack_event is destroyed by
+ * the library. Any triggered operations on the event should be completed before
+ * calling this interface. The behavior is undefined if the user acknowledges
+ * the event while waiting on the event or triggering operations on the event.
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_ack_event(ucc_ee_h ee, ucc_ev_t ev);
+
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine to set the event to the tail of the queue.
+ *
+ * @param [in]  ee        execution engine handle
+ * @param [in]  ev        Event structure fetched from the event queue
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * @ref ucc_ee_set_event sets the event on the execution engine. If the
+ * operations are waiting on the event when the user sets the event, the
+ * operations are launched. The events created by the user need to be destroyed
+ * by the user.
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_set_event(ucc_ee_h ee, ucc_ev_t ev);
+
+/**
+ * @ingroup UCC_EVENT
+ *
+ * @brief The routine blocks the calling thread until there is an event on the queue.
+ *
+ * @param [in]  ee        execution engine handle
+ * @param [out] ev        Event structure fetched from the event queue
+ *
+ * @parblock
+ *
+ * @b Description
+ *
+ * The user thread invoking the @ref ucc_ee_wait interface is blocked until an
+ * event is posted to the execution engine.
+ *
+ * @endparblock
+ *
+ * @return Error code as defined by ucc_status_t
+ */
+ucc_status_t ucc_ee_wait(ucc_ee_h ee, ucc_ev_t *ev);
```

Comment:
this function waits for a particular event or just returns first event appeared in execution enginine?

Reply:
Any event ..